examples/stripe-integration/app/routes/buy.tsx

```diff
@@ -0,0 +1,62 @@
+import { loadStripe } from "@stripe/stripe-js";
+import { Form, useActionData, useLoaderData } from "remix";
+import type { ActionFunction } from "remix";
+import { getStripeSession, getDomainUrl } from "~/utils/stripe.server";
+
+type loaderData = {
+  ENV: {
+    PRICE_ID: string;
+  };
+};
+type actionData = {
+  id: string;
+  ENV: {
+    STRIPE_PUBLIC_KEY: string;
+  };
+};
+export const loader = async (): Promise<loaderData> => {
+  return {
+    ENV: {
+      PRICE_ID: process.env.PRICE_ID as string
+    }
+  };
+};
+export const action: ActionFunction = async ({
+  request
+}): Promise<actionData> => {
+  const formData = await request.formData();
+  const id = await getStripeSession(
+    formData.get("price_id") as string,
+    getDomainUrl(request)
+  );
```

Comment:
I'm confused why this needs to be in the action function. Why can't we put this in the loader and then just have a button with an `onClick` that calls `stripe.redirectToCheckout`. Or even better, if Stripe has a way to get that URL so *we* can do the redirect we could still have a form, but we can do the redirect server side.

Reply:
Sorry Kent missed your comment somehow - Yes you are very correct! We can indeed get the stripe redirect url after getting the session. I have refractor it so now it handles all stripe functions purely on the server side.